source/guides/deployment.rst

```diff
@@ -3,24 +3,187 @@ Deploy Mattermost
+
+        .. code:: bash
+        
+            mkdir -p ./volumes/app/mattermost/{config,data,logs,plugins,client/plugins,bleve-indexes}
+            sudo chown -R 2000:2000 ./volumes/app/mattermost
+
+      4. Configure TLS for NGINX *(optional)*. If you're not using the included NGINX reverse proxy, you can skip this step.
+
+          **If creating a new certificate and key:**
+
+            .. code:: bash
+  
+                bash scripts/issue-certificate.sh -d <YOUR_MM_DOMAIN> -o ${PWD}/certs
+
+          To include the certificate and key, uncomment the following lines in your ``.env`` file and ensure they point to the appropriate files:
+
+            .. code:: bash
+  
+                #CERT_PATH=./certs/etc/letsencrypt/live/${DOMAIN}/fullchain.pem
+                #KEY_PATH=./certs/etc/letsencrypt/live/${DOMAIN}/privkey.pem
+
+          **If using a pre-existing certificate and key:**
+
+            .. code:: bash
+  
+                mkdir -p ./volumes/web/cert
+                cp <PATH-TO-PRE-EXISTING-CERT>.pem ./volumes/web/cert/cert.pem
+                cp <PATH-TO-PRE-EXISTING-KEY>.pem ./volumes/web/cert/key-no-password.pem
+
+          To include the certificate and key, ensure the following lines in your ``.env`` file points to the appropriate files:
+
+            .. code:: bash
+  
+                CERT_PATH=./volumes/web/cert/cert.pem
+                KEY_PATH=./volumes/web/cert/key-no-password.pem
+
+      5. Configure SSO with GitLab *(optional)*:
+
+        If you want to use SSO with GitLab, and you're using a self-signed certificate, you have to add the PKI chain for your authority. This is required to avoid a ``Token request failed: certificate signed by unknown authority`` error.
+
+        To add the PKI chain, uncomment the following line in your ``.env`` file and ensure it points to your ``pki_chain.pem`` file:
+
+            .. code:: bash
+  
+                # - ${GITLAB_PKI_CHAIN_PATH}:/etc/ssl/certs/pki_chain.pem:ro
+
+      6. Deploy Mattermost.
+
+          **Without using the included NGINX:**
+
+            .. code:: bash
+  
+                sudo docker-compose -f docker-compose.yml -f docker-compose.without-nginx.yml up -d
+
+          To access your new Mattermost deploy, navigate to ``http://<YOUR_MM_DOMAIN>:8065/`` in your browser.
+
+          To shut down your deployment:
+
+            .. code:: bash
+  
+                sudo docker-compose -f docker-compose.yml -f docker-compose.without-nginx.yml down
+
+          **Using the included NGINX:**
+
+            .. code:: bash
+  
+                sudo docker-compose -f docker-compose.yml -f docker-compose.nginx.yml up -d
+
+          To access your new Mattermost deploy via HTTPS, navigate to ``https://<YOUR_MM_DOMAIN>/`` in your browser.
+
+          To shut down your deployment:
+
+            .. code:: bash
+  
+                sudo docker-compose -f docker-compose.yml -f docker-compose.nginx.yml down
+    
+    .. tab:: From Tar
```

Comment:
There are a bunch of steps we need to include here. The full list is here

http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/5522/install/install-tar.html